ui/src/app/settings/components/repos-list/repos-list.tsx

```diff
@@ -254,250 +313,174 @@ export class ReposList extends React.Component<RouteComponentProps<any>, {connec
                                 className='argo-button argo-button--base'
                                 onClick={() => {
                                     this.credsTemplate = true;
-                                    this.formApiHTTPS.submitForm(null);
+                                    this.formApi.submitForm(null);
                                 }}>
                                 Save as credentials template
                             </button>{' '}
-                            <button onClick={() => (this.showConnectHTTPSRepo = false)} className='argo-button argo-button--base-o'>
+                            <button onClick={() => (this.showConnectRepo = false)} className='argo-button argo-button--base-o'>
                                 Cancel
                             </button>
-                        </div>
+                        </>
                     }>
+                    {this.ConnectRepoFormButton(this.state.method, method => {
+                        this.setState({method});
+                    })}
                     <DataLoader load={() => services.projects.list('items.metadata.name').then(projects => projects.map(proj => proj.metadata.name).sort())}>
                         {projects => (
                             <Form
-                                onSubmit={params => this.connectHTTPSRepo(params as NewHTTPSRepoParams)}
-                                getApi={api => (this.formApiHTTPS = api)}
-                                defaultValues={{type: 'git'}}
-                                validateError={(params: NewHTTPSRepoParams) => ({
-                                    url: (!params.url && 'Repository URL is required') || (this.credsTemplate && !this.isHTTPSUrl(params.url) && 'Not a valid HTTPS URL'),
-                                    name: params.type === 'helm' && !params.name && 'Name is required',
-                                    password: !params.password && params.username && 'Password is required if username is given.',
-                                    tlsClientCertKey: !params.tlsClientCertKey && params.tlsClientCertData && 'TLS client cert key is required if TLS client cert is given.'
-                                })}>
+                                onSubmit={this.onSubmitForm(this.state.method)}
+                                getApi={api => (this.formApi = api)}
+                                defaultValues={this.onChooseDefaultValues(this.state.method)}
+                                validateError={this.onValidateErrors(this.state.method)}>
```

Comment:
The onValidateError needs to get the Form params.

Have a look at the existing behavior like this:

1. Go to https://cd.apps.argoproj.io/settings/repos
2. Click on "CONNECT REPO USING HTTPS"
3. Change `Type` to `Helm`
4. Click on Connect

Notice the Name field and the URL field have errors.